discord/embeds.py

```diff
@@ -72,10 +73,21 @@ def set_thumbnail(self, *, url, height: int = None, width: int = None):
         return self
 
     def __len__(self):
-        length = len(self.title) + len(self.description)
+        length = 0
+        if self.title:
+            length += len(self.title)
+        if self.description:
+            length += len(self.description)
+
         for field in self.fields:
             length += len(field["name"]) + len(field["value"])
-        length += len(self.footer["text"]) + len(self.author["name"]) + len(self.timestamp)
+
+        if hasattr(self, "footer"):
+            length += len(self.footer["text"])
+        if hasattr(self, "author"):
+            length += len(self.author["name"])
+        if hasattr(self, "timestamp"):
+            length += len(self.timestamp) if self.timestamp else 0
```

Comment:
Why is there the extra `if self.timestamp else 0` when you already checked with hasattr?

Reply:
`self.timestamp` can be `None`

Reply:
yeah the `hasattr(self, "timestamp")` is not needed